src/core/i18n/utils.js

```diff
@@ -173,3 +175,26 @@ export function getLanguage({ lang, acceptLanguage } = {}) {
   // - normalization e.g: en-us -> en-US.
   return { lang: sanitizeLanguage(userLang), isLangFromHeader };
 }
+
+// moment uses locales like "en-gb" whereas we use "en_GB".
+export function makeMomentLocale(locale) {
+  return locale.replace('_', '-').toLowerCase();
+}
+
+// Create an i18n object with a translated moment object available we can
+// use for translated dates across the app.
+export function makeI18n(i18nData, _Jed = Jed) {
+  const i18n = new _Jed(i18nData);
+
+  // This adds the correct moment locale for the active locale so we can get
+  // localised dates, times, etc.
+  if (i18n.options && typeof i18n.options._momentDefineLocale === 'function') {
+    i18n.options._momentDefineLocale();
+    moment.locale(makeMomentLocale(i18n.options.locale_data.messages[''].lang));
```

Comment:
That is really unfortunate. I guess we could attach a function that passes the args to moment(), sets the locale and returns it.

It might not be worth the complexity but if it works first try might be worth a shot. If it doesn't work right away I'd say leave it.